management/WFCORE-2647_LDAP_SecurityRealms_Always_Send_CERT.adoc

```diff
@@ -0,0 +1,72 @@
+
+For the first connection which is used for the searches the server is either configured with it's own username and password for use when connecting to the LDAP server, or it can be configured to connect using SSL with it's own certificate.  For the connection then established to verify the remote users password we deliberately avoid using the servers credentials so we do not inadvertently authenticate as the server, in the case of an SSL connection this means we disable sending the servers certificate.
+
+For some LDAP servers it is desirable that the servers certificate is used for both of these connections, the LDAP server will then only allow connections to be established with a valid certificate AND password based authentication.
+
+The purpose of this RFE is to add a configuration option to ensure the server will always use it's certificate for all of the connections and not just for the connections performing the LDAP queries.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-2647[WFCORE-2647]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-8631[WFLY-8631]
+
+=== Dev Contacts
+
+This issue was originally developed by Peter Palaga.
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+WildFly Core - Management model and implementation.
+WildFly - Test case and documentation.
+
+=== Other Interested Projects
+
+None
+
+== Requirements
+
+=== Hard Requirements
+
+A new boolean configuration option `always-send-client-cert` should be added to the resource `core-service=management/ldap-connection=*`.
+When the new attribute is set to `true` the configured client certificate should be used for both connections to the LDAP server, i.e. the connection to search for the authenticating identity and the follow up connection to verify the password supplied for that identity.
+When the new attribute is set to `false` the configured client certificate should only be used on connections searching LDAP, the certificate should not be used on connections verifying the identities supplied password.
+The new attribute should default to `false` to ensure consistent behaviour from any configuration being ported from a prior version of the server. 
+
+Prior to this RFE this behaviour had already been added to a maintenance or release of JBoss EAP but configurable using the system property `jboss.as.management.outbound.ldap.alwaysSendClientCert`, where the new attribute is not specified on the  `ldap-connection` resource the system property should continue to be usable to allow migrations of existing configuration.
```

Comment:
@darranl based on our yesterday discussion, maybe add note this system property should be removed in future - 2 releases after proper model handling (WF14) - WF 16

Reply:
Done, but I would not worry if it lives on a little bit, this entire resource is up for removal a few CDs later.